src/github-handler/comment-handler/suggestion-patch-handler/index.ts

```diff
@@ -0,0 +1,56 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {getRawSuggestionHunks} from './suggestion-hunk-handler';
+import {
+  filterOutOfScopeHunks,
+  filterOutOfScopeFiles,
+} from './fix-suggestion-hunks-handler';
+import {RawChanges, FileHunks, FileRanges, FilePatches} from '../../../types';
+
+/**
+ * Get the valid hunks and files
+ * @param {RawChanges} rawChanges the raw old content and new content of a file
+ * @param {string[]} invalidFiles list of invalid files
+ * @param {FileRanges} validFileLines a map of each file's valid lines for a Pull Request
+ */
+export function getValidSuggestionHunks(
+  rawChanges: RawChanges,
+  invalidFiles: string[],
+  validFileLines: FileRanges
+): FileHunks {
+  const fileHunks = getRawSuggestionHunks(rawChanges);
+  filterOutOfScopeFiles(invalidFiles, fileHunks);
+  filterOutOfScopeHunks(validFileLines, fileHunks);
+  return fileHunks;
```

Comment:
refactor option for future: could this use `Array.filter()`?

something like:
```suggestion
  return fileHunks = getRawSuggestionHunks(rawChanges)
    .filter(hunk => !outOfScopeFile(invalidFiles, hunk) && !outOfScopeHunks(validFileLines));
```

Reply:
I don't think we can do that. The current `fileHunk` is a map so the only option is to produce a new map, or edit the old map which is the current solution. As mentioned earlier, we also need a map to do raw text lookups for particular files.